### ISSUE TITLE
eval

### ISSUE THREAD

**Issue by mc-lgt** [OpenGVLab/InternImage#8]
Hello, I use internimage as my model backone. Why does the training loss drop normally, but the map in the verification set is all 0

**Comment by czczup** [comment]
> Hello, I use internimage as my model backone. Why does the training loss drop normally, but the map in the verification set is all 0

hello, could you provide more information.

**Comment by mc-lgt** [comment]
I use the T version, and the amount of parameters and calculation is not high during training, but the training time is more than twice that of using the swing transformer as the backbone

**Comment by mc-lgt** [comment]
Is it related to the convolution operator?

**Comment by czczup** [comment]
> Is it related to the convolution operator?

I just tested the training time, in my machine (8 A100), Swin-T takes 0.23s/iter, and InternImage takes 0.34s/iter. 
It is slower than Swin-T because InternImage-T is more deeper.

May I ask if you train InternImage-T on COCO, or other datasets?

**Comment by mc-lgt** [comment]
I use 3080 in my own defect data set

**Comment by czczup** [comment]
> I use 3080 in my own defect data set

Have you compiled the DCNv3 CUDA operator successfully? And, what's your version of pytorch or CUDA.
This information may help me identify the problem.

I am training InternImage-T on COCO again, but it gives me a normal mAP at epoch 1.

**Comment by mc-lgt** [comment]
Result of running test file:
>>> forward double
* True check_forward_equal_with_pytorch_double: max_abs_err 8.16e-10 max_rel_err nan
>>> forward float
* True check_forward_equal_with_pytorch_float: max_abs_err 1.75e-09 max_rel_err nan
>>> backward double: channels 1
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 1.19e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 1.51e-09 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 3.26e-09 max_rel_err nan
>>> backward double: channels 16
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 1.49e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 1.02e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 4.34e-08 max_rel_err nan
>>> backward double: channels 30
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 1.34e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 1.86e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 8.20e-08 max_rel_err nan
>>> backward double: channels 32
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 1.04e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 2.79e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 9.50e-08 max_rel_err nan
>>> backward double: channels 64
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 1.04e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 1.23e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 1.79e-07 max_rel_err nan
>>> backward double: channels 71
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 8.94e-08 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 2.24e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 1.49e-07 max_rel_err nan
>>> backward double: channels 1025
* True input_grad check_backward_equal_with_pytorch_double: max_abs_err 1.64e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_double: max_abs_err 7.75e-07 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_double: max_abs_err 2.38e-06 max_rel_err nan
>>> backward float: channels 1
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 1.49e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 2.68e-09 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 9.31e-09 max_rel_err nan
>>> backward float: channels 16
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 2.38e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 7.45e-09 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 1.04e-07 max_rel_err nan
>>> backward float: channels 30
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 1.79e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 4.56e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 1.64e-07 max_rel_err nan
>>> backward float: channels 32
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 3.28e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 5.22e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 1.64e-07 max_rel_err nan
>>> backward float: channels 64
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 2.98e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 4.47e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 3.58e-07 max_rel_err nan
>>> backward float: channels 71
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 2.68e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 8.20e-08 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 3.73e-07 max_rel_err nan
>>> backward float: channels 1025
* True input_grad check_backward_equal_with_pytorch_float: max_abs_err 1.79e-07 max_rel_err nan
* True offset_grad check_backward_equal_with_pytorch_float: max_abs_err 2.74e-06 max_rel_err nan
* True mask_grad check_backward_equal_with_pytorch_float: max_abs_err 7.15e-06 max_rel_err nan
>>> time cost: im2col_step 128; input torch.Size([512, 64, 64, 64]); points 9 
foward time cost: 0.010530190467834473
>>> time cost: im2col_step 256; input torch.Size([512, 64, 64, 64]); points 9 
foward time cost: 0.010517220497131347
>>> time cost: im2col_step 512; input torch.Size([512, 64, 64, 64]); points 9 
foward time cost: 0.010549230575561524

Process finished with exit code 0

**Comment by czczup** [comment]
In addition, please check if you use deterministic as True in your code:
That means set:
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
I noticed that these will obvious slow down the training speed of InternImage.

**Comment by czczup** [comment]
> could you provide me the log of loading pretrained weights, such as missing keys.

Thanks, that's no problem.
Could you provide me the log of loading pretrained weights, such as missing keys.

**Comment by mc-lgt** [comment]
using core type: DCNv3
using activation layer: GELU
using main norm layer: LN
using dpr: linear, 0.2
<All keys matched successfully>

**Comment by mc-lgt** [comment]
using core type: DCNv3
using activation layer: GELU
using main norm layer: LN
using dpr: linear, 0.2
<All keys matched successfully>

**Comment by czczup** [comment]
> using core type: DCNv3 using activation layer: GELU using main norm layer: LN

That's right.

Have you see a log like this:
```
2023-03-05 14:42:30,032 - mmdet - INFO - load checkpoint from http path: https://github.com/OpenGVLab/InternImage/releases/download/cls_model/internimage_t_1k_224.pth
2023-03-05 14:42:30,309 - mmdet - INFO - _IncompatibleKeys(missing_keys=['levels.0.blocks.0.gamma1', 'levels.0.blocks.0.gamma2', 'levels.0.blocks.1.gamma1', 'levels.0.blocks.1.gamma
2', 'levels.0.blocks.2.gamma1', 'levels.0.blocks.2.gamma2', 'levels.0.blocks.3.gamma1', 'levels.0.blocks.3.gamma2', 'levels.1.blocks.0.gamma1', 'levels.1.blocks.0.gamma2', 'levels.1
.blocks.1.gamma1', 'levels.1.blocks.1.gamma2', 'levels.1.blocks.2.gamma1', 'levels.1.blocks.2.gamma2', 'levels.1.blocks.3.gamma1', 'levels.1.blocks.3.gamma2', 'levels.2.blocks.0.gam
ma1', 'levels.2.blocks.0.gamma2', 'levels.2.blocks.1.gamma1', 'levels.2.blocks.1.gamma2', 'levels.2.blocks.2.gamma1', 'levels.2.blocks.2.gamma2', 'levels.2.blocks.3.gamma1', 'levels
.2.blocks.3.gamma2', 'levels.2.blocks.4.gamma1', 'levels.2.blocks.4.gamma2', 'levels.2.blocks.5.gamma1', 'levels.2.blocks.5.gamma2', 'levels.2.blocks.6.gamma1', 'levels.2.blocks.6.g
amma2', 'levels.2.blocks.7.gamma1', 'levels.2.blocks.7.gamma2', 'levels.2.blocks.8.gamma1', 'levels.2.blocks.8.gamma2', 'levels.2.blocks.9.gamma1', 'levels.2.blocks.9.gamma2', 'leve
ls.2.blocks.10.gamma1', 'levels.2.blocks.10.gamma2', 'levels.2.blocks.11.gamma1', 'levels.2.blocks.11.gamma2', 'levels.2.blocks.12.gamma1', 'levels.2.blocks.12.gamma2', 'levels.2.bl
ocks.13.gamma1', 'levels.2.blocks.13.gamma2', 'levels.2.blocks.14.gamma1', 'levels.2.blocks.14.gamma2', 'levels.2.blocks.15.gamma1', 'levels.2.blocks.15.gamma2', 'levels.2.blocks.16
.gamma1', 'levels.2.blocks.16.gamma2', 'levels.2.blocks.17.gamma1', 'levels.2.blocks.17.gamma2', 'levels.3.blocks.0.gamma1', 'levels.3.blocks.0.gamma2', 'levels.3.blocks.1.gamma1',
'levels.3.blocks.1.gamma2', 'levels.3.blocks.2.gamma1', 'levels.3.blocks.2.gamma2', 'levels.3.blocks.3.gamma1', 'levels.3.blocks.3.gamma2'], unexpected_keys=['conv_head.0.weight', '
conv_head.1.0.weight', 'conv_head.1.0.bias', 'conv_head.1.0.running_mean', 'conv_head.1.0.running_var', 'conv_head.1.0.num_batches_tracked', 'head.weight', 'head.bias'])
```

**Comment by mc-lgt** [comment]
No, because I didn't use the mmdet code

**Comment by mc-lgt** [comment]
I just set torch.backends.cudnn.deterministic to False, and the training time has been reduced by more than half

**Comment by czczup** [comment]
> I just set torch.backends.cudnn.deterministic to False, and the training time has been reduced by more than half

Okay. For the abnormal val mAP, may I suggest that you check if the pretrained checkpoint has been loaded successfully?

**Comment by mc-lgt** [comment]
Ok, let me see again. Thank you for your answer

**Comment by czczup** [comment]
> Ok, let me see again. Thank you for your answer

<img width="420" alt="image" src="https://user-images.githubusercontent.com/23737120/222948290-6a6de538-ba33-40fc-898c-1cd6fc79eafd.png">

**Comment by czczup** [comment]
Closing this issue as there has been no activity. Please feel free to reopen if required. Thanks for your feedback and support.
